.github/workflows/build_and_test.yml

```diff
@@ -32,22 +33,33 @@ jobs:
           extra_cmake_args: ""
     container:
       image: rostooling/setup-ros-docker:ubuntu-${{ matrix.ubuntu_distro }}-ros-${{ matrix.ros_distro }}-ros-base-latest
-    outputs:
-      build_result: ${{ steps.build.outcome }}
-      test_result: ${{ steps.test.outcome }}
+      env:
+        # Needed for the CMakeLists.txt setup
+        ROS_DISTRO: ${{ matrix.ros_distro }}
+        ROS_VERSION: ${{ matrix.ros_version }}
     steps:
     # TODO(setup-ros-docker#7): calling chown is necessary for now
-    - run: sudo chown -R rosbuild:rosbuild "$HOME" .
-    # Needed for the CMakeLists.txt setup
-    - run: export ROS_DISTRO=${{ matrix.ros_distro }}
-    # Needed for the CMakeLists.txt setup
-    - run: export ROS_VERSION=${{ matrix.ros_version }}
-    - uses: ros-tooling/action-ros-ci@0.0.17
+    - name: setup-ros-docker workaround
+      run: sudo chown -R rosbuild:rosbuild "$HOME" .
+    # Needed to access the vcs repos file from the workspace
+    - name: Checkout source
+      uses: actions/checkout@v2
+    - name: Build and test ${{ matrix.ros_distro }} ${{ matrix.ros_version }} ${{ matrix.build_type }}
+      uses: ros-tooling/action-ros-ci@0.0.17
+      if: ${{ matrix.build_type == 'master' }}
       with:
         source-ros-binary-installation: ${{ matrix.ros_distro }}
         package-name: aws_common
         extra-cmake-args: ${{ matrix.extra_cmake_args }}
-        vcs-repo-file-url: ''
+        vcs-repo-file-url: ""
+    - name: Build and test ${{ matrix.ros_distro }} ${{ matrix.ros_version }} ${{ matrix.build_type }}
+      uses: ros-tooling/action-ros-ci@0.0.17
+      if: ${{ matrix.build_type == 'release_latest' }}
+      with:
+        source-ros-binary-installation: ${{ matrix.ros_distro }}
+        package-name: aws_common
+        extra-cmake-args: ${{ matrix.extra_cmake_args }}
+        vcs-repo-file-url: "${{ github.workspace }}/.github/workflows/release_latest.repos"
```

Comment:
i was wondering if it was possible to conditionally set this variable via an expression, something like 
```
vcs-repo-file-url: ${{ "" if matrix.build_branch == "master" else "{}/.github/workflows/release_latest.repos".format(github.workspace) }}
```

but i'm not familiar enough with workflow expression syntax to know if that is expressible (above is python syntax which probably isn't right)

If not expressible then this looks like a good approach - much less duplication than before :+1:

Reply:
Ah - not directly expressible but there is a way to approach it https://github.community/t/do-expressions-support-ternary-operators-to-change-their-returned-value/18114/3 

Could set an output/env var conditionally (it defaulting to empty/null in the other case). Anyways, still optional

Reply:
I don't think a ternary would work either given how the upload needs to be conditional as well.